crypto/rsa/rsa_gen.c

```diff
@@ -234,15 +233,11 @@ static int rsa_builtin_keygen(RSA *rsa, int bits, int primes, BIGNUM *e_value,
          * distinguish a multi-prime private key by using the modulus
          * in a certificate.
          */
-        nlb = (nl + 7) / 8;
-        bin = OPENSSL_malloc(nlb);
-        if (bin == NULL)
+        if (!BN_rshift(r2, r1, nl - 4))
```

Comment:
Yes, this should work, you can try to do
`BN_rshift(r2, r1, bitse-4)` here...

Reply:
bitse is not the actual length of r1, so if you want to check the top 4 bits of r1, nl should be used instead.